lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -816,6 +816,26 @@ func (c *Client) sharedDirectoryMoveRequest(req tdp.SharedDirectoryMoveRequest)
 // the TDP connection to the browser.
 func (c *Client) close() {
 	c.closeOnce.Do(func() {
+		// In the case that the session ends due to an RDP server disconnect,
+		// the TDP connection will still be open at this point. Therefore
+		// we can ask for the RDP server disconnect reason here and send it
+		// back to the user via TDP.
+		res := C.get_server_disconnect_reason(c.rustClient)
+		if res.err != C.ErrCodeSuccess {
+			c.cfg.Log.Errorf("error getting server disconnect reason: %v", res.err)
+		} else {
+			reason := C.GoString(res.reason)
+			C.free_c_string(res.reason)
+			if reason != "" {
```

Comment:
nit: similarly as above here?

```suggestion
			if reason == "" {
				return			
			}
```

Reply:
Same thing here -- we don't actually want to return early in this particular context.